src/cls/IPM/Utils/Module.cls

```diff
@@ -2288,19 +2290,47 @@ ClassMethod %EvaluateSystemExpression(pString As %String) As %String [ Internal
 
 	Set dbRole = ..GetDatabaseRole()
 	Set result = ..%RegExReplace(result, "dbrole", dbRole)
-
-	Set result = ..%RegExReplace(result, "authK5API", $$$AutheK5API)
-	Set result = ..%RegExReplace(result, "authPassword", $$$AutheCache)
-	Set result = ..%RegExReplace(result, "authUnauthenticated", $$$AutheUnauthenticated)
-	Set result = ..%RegExReplace(result, "authLDAP", $$$AutheLDAP)
-	Set result = ..%RegExReplace(result, "authDelegated", $$$AutheDelegated)
-	Set result = ..%RegExReplace(result, "authLoginToken", $$$AutheLoginToken)
-	Set result = ..%RegExReplace(result, "authTwoFactorSMS", $$$AutheTwoFactorSMS)
-	Set result = ..%RegExReplace(result, "authTwoFactorPW", $$$AutheTwoFactorPW)
+	Set result = ..%RegExReplace(result, "globalsDbRole", dbRole)
 
 	Return result
 }
 
+ClassMethod %EvaluateMacro(pString As %String) As %String [ Internal ]
+{
+	Set tString = pString
+	Set tPattern = "\$\$\$[a-zA-Z0-9]+"
+	Set tMatch = ##class(%Regex.Matcher).%New(tPattern, tString)
+	While tMatch.Locate() {
+		Set group = tMatch.Group
+		Set macroVal = ..EvaluateMacroHelper($Extract(group, 4, *))
+		set tString = $Replace(tString, group, macroVal)
+	}
+	Return tString
+}
+
+/// Helper metohd to evaluate macro at runtime. 
+ClassMethod EvaluateMacroHelper(pMacro As %String) As %String [ Internal, Private ]
+{
+    Set stream = ##class(%Stream.TmpCharacter).%New()
+    Do stream.WriteLine("Class %IPM.Generated.MacroEval")
+    Do stream.WriteLine("{")
+    Do stream.WriteLine("ClassMethod EvalMacro() [ CodeMode = objectgenerator ]")
+    Do stream.WriteLine("{")
+    Do stream.WriteLine("  Do %code.WriteLine(""#IfDef "_ pMacro _""")")
+    Do stream.WriteLine("  Do %code.WriteLine("" Quit $$$"_pMacro_""")")
+    Do stream.WriteLine("  Do %code.WriteLine(""#Else "")")
+    Do stream.WriteLine("  Do %code.WriteLine("" Quit """""""" "")")
+    Do stream.WriteLine("  Do %code.WriteLine(""#EndIf "")")
+    Do stream.WriteLine("  Quit $$$OK") 
+    Do stream.WriteLine("}")
+    Do stream.WriteLine("}")
+
+    Do $system.OBJ.LoadStream(stream, "ck-d")
```

Comment:
Check if there was a compilation error by checking the status here and if so, don't call the class method below